pyxis/create_container_image.py

```
@@ -0,0 +1,156 @@
+        "--skopeo-result",
+        help="File with result of `skopeo inspect` running against image"
+        " represented by ContainerImage to be created",
+        required=True,
+    )
+    parser.add_argument("--is-latest", help="Should the `latest` tag of the ContainerImage be overwritten?", required=True)
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def check_if_image_already_exists(args, digest: str) -> bool:
+    # quote is needed to urlparse the quotation marks
+    filter_str = quote(
+        f'docker_image_digest=="{digest}";'
+        f"not(deleted==true)"
+    )
+
+    check_url = urljoin(args.pyxis_url, f"v1/images?page_size=1&filter={filter_str}")
+
+    # Get the list of the ContainerImages with given parameters
+    rsp = pyxis.get(check_url)
+    rsp.raise_for_status()
+
+    query_results = rsp.json()["data"]
+
+    if len(query_results) == 0:
+        LOGGER.info(
+            "Image with given docker_image_digest doesn't exist yet"
+        )
+        return False
+
+    LOGGER.info(
+        "Image with given docker_image_digest already exists."
+        "Skipping the image creation."
+    )
+    return True
+
+
+def prepare_parsed_data(skopeo_result: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "digest": skopeo_result.get("Digest", ""),
+        "docker_version": skopeo_result.get("DockerVersion", ""),
+        "layers": skopeo_result.get("Layers", []),
+        "name": skopeo_result.get("Name", ""),
+        "architecture": skopeo_result.get("Architecture", ""),
+        "env_variables": skopeo_result.get("Env", []),
+    }
+
+
+def create_container_image(
+    args,
+    parsed_data : Dict[str, Any],
+):
+    LOGGER.info("Creating new container image")
+
+    date_now = datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%f+00:00")
+
+    # digest and name aren't accepted in the parsed_data payload to pyxis
+    docker_image_digest = parsed_data["digest"]
```

Are you ok with the script failing on runtime if the key doesn't exist?

---

Yes, it should fail and do nothing if the key doesn't exist. Do you think I should raise a custom exception saying as such in this case, or just leave it and let the normal python failure tell the story?

---

Probably a cleaner error would be better, but I'm also ok with leaving it like this. Your call :wink:

---

fixed